src/scipp/typing.py

```diff
@@ -87,3 +87,16 @@ def has_numeric_type(obj: _std_typing.Any) -> bool:
   - :class:`type` objects like :class:`int` and :class:`float`
   - names of dtypes as strings like ``'int32'`` and ``'float64'``
 """  # noqa: E501
+
+if _std_typing.TYPE_CHECKING:
+    from enum import Enum
+
+    class ellipsis(Enum):
+        Ellipsis = "..."
```

Comment:
What is this good for?

Reply:
You can use it to update values but keeping the same buffer, with unknown dims:

```python
x = sc.arange('x', 4)
da = sc.DataArray(x)
da.data[...] = 2*x
```

Reply:
No, I mean why do you distinguish between type checking and running the code?

Reply:
I cannot remember, I copied this from scippnexus, but this was the only way to get it to work.